mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -193,12 +207,19 @@ class KMeans private (
 
     val initStartTime = System.nanoTime()
 
-    val centers = if (initializationMode == KMeans.RANDOM) {
-      initRandom(data)
-    } else {
-      initKMeansParallel(data)
+    val centers = initialModel match {
+      case Some(kMeansCenters) => {
+        Array.tabulate(runs)(r => kMeansCenters.clusterCenters
```

Comment:
I think we should:
* Not change the value of runs, but ignore it when initialModel is given
* Document this behavior in the Scala/Python docs for runs and initialModel
* Add a logWarning which prints a warning message when runs is being ignored